src/python/pants/backend/python/tasks/python_bundle.py

```diff
@@ -0,0 +1,86 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+
+from pants.backend.python.targets.python_app import PythonApp
+from pants.base.build_environment import get_buildroot
+from pants.base.exceptions import TargetDefinitionException
+from pants.fs import archive
+from pants.task.task import Task
+from pants.util import dirutil
+
+
+class PythonBundle(Task):
+  """Create an archive bundle of PythonApp targets."""
+
+  _DEPLOYABLE_ARCHIVES = 'deployable_archives'
+  _PEX_ARCHIVES = 'pex_archives'
+
+  @classmethod
+  def product_types(cls):
+    return [cls._DEPLOYABLE_ARCHIVES]
+
+  @classmethod
+  def prepare(cls, options, round_manager):
+    super(PythonBundle, cls).prepare(options, round_manager)
+    round_manager.require_data(cls._PEX_ARCHIVES)
+
+  def __init__(self, *args, **kwargs):
+    super(PythonBundle, self).__init__(*args, **kwargs)
+    self._outdir = self.get_options().pants_distdir
+
+  def execute(self):
+    targets_to_bundle = self.context.targets(PythonApp.is_python_app)
+    dirutil.safe_mkdir(self._outdir)
+    for target in targets_to_bundle:
+      bundle_dir = os.path.join(self.workdir, target.id)
+      binary_path = self._get_binary_path(target)
+      dirutil.safe_mkdir(bundle_dir, clean=True)
+      os.symlink(binary_path, os.path.join(bundle_dir, os.path.basename(binary_path)))
+      self._symlink_bundles(target, bundle_dir)
+      archiver = archive.archiver(target.payload.archive)
+      archivepath = archiver.create(
+        bundle_dir,
+        self._outdir,
+        target.name,
+      )
+      bundle_archive_product = self.context.products.get(self._DEPLOYABLE_ARCHIVES)
+      bundle_archive_product.add(
+        target, os.path.dirname(archivepath)).append(os.path.basename(archivepath))
+      self.context.log.info(
+        'created {}'.format(os.path.relpath(archivepath, get_buildroot())))
+
+  def _get_binary_path(self, target):
+    pex_archives = self.context.products.get(self._PEX_ARCHIVES)
+    for basedir, filenames in pex_archives.get(target.binary).items():
+      for filename in filenames:
+        return os.path.join(basedir, filename)
+
+  def _symlink_bundles(self, app, bundle_dir):
```

Comment:
Can this be moved out as a static function somewhere? Or would it be useful to have a `BundleTaskMixin` to provide it?